WooCommerce/Classes/Tools/AggregateData/AggregateDataHelper.swift

```diff
@@ -131,7 +120,9 @@ final class AggregateDataHelper {
             )
         }
 
-        let sorted = unsortedResult.sorted(by: { ($0.productID, $0.variationID) < ($1.productID, $1.variationID) })
+        let filtered = unsortedResult.filter { $0.quantity > 0 }
```

Comment:
nit: we could add a unit test case to verify the products with quantity 0 are filtered :)